dev/documentation/en/utilities-email.md

```diff
@@ -0,0 +1,267 @@
+
+// [
+//     'journalName' => 'Journal of Public Knowledge',
+//     'journalUrl' => 'https://journal.com',
+//     'submissionTitle' => 'Yam diseases and its management in Nigeria',
+//     'submissionUrl' => 'https://journal.com/workflow/access/15',
+//     ...
+// ];
+```
+
+Use the variables in an email's subject and body.
+
+```php
+use APP\facades\Repo;
+use Illuminate\Support\Facades\Mail;
+use PKP\mail\mailables\SubmissionReceived;
+
+$subject = 'New submission to {$journalName}';
+$body = 'A new submission is ready to review: {$submissionTitle}.';
+
+$mailable = new SubmissionReceived($context, $submission);
+$mailable->from('nate@example.com')
+    ->to('vitaliy@example.com')
+    ->subject($subject)
+    ->body($body);
+
+Mail::send($mailable);
+```
+
+All variables assigned to a Mailable can be retrieved from the `viewData` property.
+
+```php
+use PKP\mail\mailables\SubmissionReceived;
+
+$mailable = new SubmissionReceived($context, $submission);
+echo $mailable->viewData['journalName'];
+
+// outputs: Journal of Public Knowledge
+```
+
+Not all objects passed to a Mailable's constructor will produce email variables. The supported classes are defined at `Mailable::templateVariablesMap()`.
+
+### Sender and Recipients
+
+Mailables provide common variables for the sender and recipients, such as `{$senderName}` and `{$recipientsName}`. Add the `Sender` or `Recipient` traits to a Mailable class to use these variables.
+
+```php
+namespace PKP\mail\mailables;
+
+use PKP\mail\Mailable;
+use PKP\mail\mailables\Recipient;
+use PKP\mail\mailables\Sender;
+
+class SubmissionReceived extends Mailable
+{
+    use Recipient;
+    use Sender;
+
+    /* ... */
+}
+```
+
+Use the `setRecipients()` and `setSender()` methods instead of the `from()` and `to()` methods.
+
+```php
+use APP\facades\Repo;
+use Illuminate\Support\Facades\Mail;
+use PKP\mail\mailables\SubmissionReceived;
+
+$recipients = Repo::users()->getMany($collector);
+
+$mailable = new SubmissionReceived($context, $submission);
+$mailable->setSender($request->getUser())
+    ->setRecipients($recipients)
+    ->subject($subject)
+    ->body($body);
+
+Mail::send($mailable);
+```
+
+### Custom Variables
+
+A Mailable may require variables that can not be generated automatically from the dependencies defined in the constructor. These variables can be passed to the Mailable before it is sent.
+
+```php
+use Illuminate\Support\Facades\Mail;
+use PKP\mail\mailables\SubmissionReceived;
+
+$mailable = new SubmissionReceived($context, $submission);
+
+$mailable->addVariables([
+        'section' => $section->getLocalizedTitle(),
+    ])
+    ->setSender($request->getUser())
+    ->setRecipients($recipients)
+    ->subject($subject)
+    ->body('A submission was made to {$section}.');
+
+Mail::send($mailable);
+```
+
+Variables added in this way need to be added to the Mailable class, so that an editor will know this variable can be used when they customize the email template.
```

Comment:
I don't have any practical examples of this in the code yet. I will keep in mind to review this part of the docs later.

Reply:
Does the example below not cover it? I have extended the `getVariables()` method to document the custom variable.

Reply:
I meant real-world examples in the current codebase. Nevertheless, it should be ok.